arch/x86_64/mcount-insn.c

```diff
@@ -193,6 +193,209 @@ static int handle_pic_lea(cs_insn *insn, uint8_t insns[],
 	return -1;
 }
 
+/*
+ *  handle PIC code for MOV instruction.
+ *  this function manipulate the instruction.
+ *
+ *  for examples :
+ *
+ *  case LOAD:
+ *	mov rcx, qword ptr [rip + 0x8f3f85]
+ *  to manipulate like below:
+ *	push rax
+ *	mov rax, PC_BASE + 0x8f3f85
+ *	mov rcx, [rax]
+ *	pop rax
+ *
+ *  case STORE:
+ *  	mov qword ptr [rip+0x8f3f85], rcx
+ *  to manipulate like below:
+ *	push rax
+ *	mov rax, PC_BASE + 0x8f3f85
+ *	mov [rax], rcx
+ *	pop rax
+ *
+ *  exceptional case in LOAD :
+ *  	mov rax, qword ptr [rip+0x8f3f85]
+ *  this will change like below:
+ * 	xchg rax, rcx
+ *  	push rax
+ *  	mov rax, PC_BASE + 0x8f3f85
+ *  	mov rcx, [rax]
+ *  	pop rax
+ *  	xchg rax, rcx
```

Comment:
Why is this exceptional?  Why not doing below?  (But I still think we can do this just using single register like above)

```
push rcx
mov rcx, PC_BASE + 0xf8f385
mov rax, [rcx]
pop rcx
```

Reply:
@namhyung because only allowed 8byte displacement to RAX register. so, we use RAX to store address of PIC. therefore, need to keep the value of rax register if instruction which we want to manipulate use rax.